amethyst_gltf/src/format/mod.rs

```diff
@@ -0,0 +1,545 @@
+                    .collect(),
+            })
+            .ok_or(GltfError::MissingPositions)?;
+
+        let colors = primitive
+            .colors_rgba_f32(0, 1., buffers)
+            .map(|colors| match faces {
+                Some(ref faces) => {
+                    let colors = colors.collect::<Vec<_>>();
+                    faces
+                        .iter()
+                        .map(|i| Separate::<Color>::new(colors[*i]))
+                        .collect()
+                }
+                None => colors.map(|color| Separate::<Color>::new(color)).collect(),
+            });
+
+        let tex_coord = match primitive.tex_coords_f32(0, buffers) {
+            Some(tex_coords) => Some(tex_coords.collect::<Vec<[f32; 2]>>()),
+            None => match options.generate_tex_coords {
+                Some((u, v)) => Some((0..positions.len()).map(|_| [u, v]).collect()),
+                None => None,
+            },
+        }.map(|texs| match faces {
+            Some(ref faces) => faces
+                .iter()
+                .map(|i| Separate::<TexCoord>::new(texs[*i]))
+                .collect(),
+            None => texs.into_iter()
+                .map(|t| Separate::<TexCoord>::new(t))
+                .collect(),
+        });
+
+        let normals = primitive.normals(buffers).map(|normals| match faces {
+            Some(ref faces) => {
+                let normals = normals.collect::<Vec<_>>();
+                faces
+                    .iter()
+                    .map(|i| Separate::<Normal>::new(normals[*i]))
+                    .collect()
+            }
+            None => normals.map(|n| Separate::<Normal>::new(n)).collect(),
+        });
+
+        let tangents = primitive.tangents(buffers).map(|tangents| match faces {
+            Some(ref faces) => {
+                let tangents = tangents.collect::<Vec<_>>();
+                faces
+                    .iter()
+                    .map(|i| {
+                        Separate::<Tangent>::new(
+                            [tangents[*i][0], tangents[*i][1], tangents[*i][2]],
+                        )
+                    })
+                    .collect()
+            }
+            None => tangents
+                .map(|t| Separate::<Tangent>::new([t[0], t[1], t[2]]))
+                .collect(),
+        });
+
+        let material = primitive.material().index();
+        // TODO: joint ids and weights
+
+        match map_mode(primitive.mode()) {
+            Ok(primitive) => primitives.push(GltfPrimitive {
+                primitive,
+                indices: faces,
+                material,
+                attributes: (positions, colors, tex_coord, normals, tangents),
+                handle: None,
+            }),
+            Err(err) => return Err(err),
+        }
+    }
+    Ok(primitives)
+}
+
+fn map_mode(mode: gltf::json::mesh::Mode) -> Result<Primitive, GltfError> {
+    use gltf::json::mesh::Mode::*;
```

Comment:
I suggest `gltf::mesh::Mode` instead. The `gltf::json` module is deprecated and will be removed in 1.0.

Reply:
Oh, I must have missed that, it did feel weird to have to go down to the json stuff.